R/read_csv.R

```diff
@@ -204,41 +206,44 @@ read_cmdstan_csv <- function(files,
       all_draws <- 1
     }
 
+    vroom_args <- list(
+      file = output_file,
+      comment = "#",
+      delim = ",",
+      trim_ws = TRUE,
+      altrep = FALSE,
+      progress = FALSE,
+      skip = metadata$lines_to_skip,
+      col_select = col_select
+    )
     if (metadata$method == "generate_quantities") {
-      # set the first arg as double
-      # to silence the type detection info
-      col_types <- list()
-      col_types[[col_select[1]]] = "d"
-      suppressWarnings(
-        draws <- vroom::vroom(
-          output_file,
-          comment = "#",
-          delim = ',',
-          col_select = col_select,
-          col_types = col_types,
-          trim_ws = TRUE,
-          altrep = FALSE,
-          progress = FALSE,
-          skip = metadata$lines_to_skip
-        )
-      )
+      # set the first arg as double to silence the type detection info
+      vroom_args$col_types <- list()
+      vroom_args$col_types[[col_select[1]]] <- "d"
     } else {
-      suppressWarnings(
-        draws <- vroom::vroom(
-          output_file,
-          comment = "#",
-          delim = ',',
-          trim_ws = TRUE,
-          col_select = col_select,
-          col_types = c("lp__" = "d"),
-          altrep = FALSE,
-          progress = FALSE,
-          skip = metadata$lines_to_skip,
-          n_max = all_draws * 2
+      vroom_args$col_types <- c("lp__" = "d")
+      vroom_args$n_max <- all_draws * 2
+    }
+
+    draws <- try(silent = TRUE, expr = {
+      suppressWarnings(do.call(vroom::vroom, vroom_args))
+    })
+    if (inherits(draws, "try-error")) {
+      if (vroom_warnings == 0) { # only warn the first time instead of for every csv file
```

Comment:
Checking that we haven't yet thrown the warning let's us warn once instead of repeating the warning every csv file